src/main/java/com/cloudbees/jenkins/plugins/bitbucket/hooks/WebhookAutoRegisterListener.java

```diff
@@ -151,21 +169,42 @@ private synchronized void registerHooks(SCMSourceOwner owner) throws IOException
                             bitbucket.registerCommitWebHook(existing);
                         }
                     } else  if (existing == null) {
-                            LOGGER.info(String.format("Registering hook for %s/%s", source.getRepoOwner(), source.getRepository()));
-                            bitbucket.registerCommitWebHook(getHook(source));
+                        LOGGER.info(String.format("Registering hook for %s/%s", source.getRepoOwner(),
+                                source.getRepository()));
+                        bitbucket.registerCommitWebHook(getHook(source));
                     }
                 }
             }
         } else {
-            LOGGER.warning(String.format("Can not register hook. Jenkins root URL is not valid: %s", rootUrl));
+            // only complain about being unable to register the hook if someone wants the hook registered.
+            SOURCES:
+            for (BitbucketSCMSource source : sources) {
+                switch (new BitbucketSCMSourceContext(null, SCMHeadObserver.none())
+                        .withTraits(source.getTraits())
+                        .webhookRegistration()) {
+                    case DISABLE:
+                        continue SOURCES;
+                    case SYSTEM:
+                        AbstractBitbucketEndpoint endpoint =
```

Comment:
ok, so you read it here, but I haven't really seen a place where it can be set to `SYSTEM`?